### DISCUSSION TITLE
How to fix gap where background color is missing?

### DISCUSSION THREAD

**Discussion by mikehaertl** [erikw/tmux-powerline#436] [Q&A] [answered]
I have a gap on the right side where the background color is missing:

![image](https://github.com/user-attachments/assets/3511fedd-a4dd-4f20-86b6-ed20577bfa56)

I couldn't find a relevant configuration. In the `default.sh` theme file a `TMUX_POWERLINE_DEFAULT_BACKGROUND_COLOR` is set. But that's only used for the middle part (the window list).

So how can I get rid of that gap?

Here's what I changed in the default config:

```bash
export TMUX_POWERLINE_DEBUG_MODE_ENABLED="false"
export TMUX_POWERLINE_PATCHED_FONT_IN_USE="true"

export TMUX_POWERLINE_THEME="default"
export TMUX_POWERLINE_DIR_USER_THEMES="${XDG_CONFIG_HOME:-$HOME/.config}/tmux-powerline/themes"
export TMUX_POWERLINE_DIR_USER_SEGMENTS="${XDG_CONFIG_HOME:-$HOME/.config}/tmux-powerline/segments"

export TMUX_POWERLINE_STATUS_VISIBILITY="on"
export TMUX_POWERLINE_WINDOW_STATUS_LINE=0
export TMUX_POWERLINE_STATUS_INTERVAL="1"
export TMUX_POWERLINE_STATUS_JUSTIFICATION="left"

export TMUX_POWERLINE_STATUS_LEFT_LENGTH="30"
export TMUX_POWERLINE_STATUS_RIGHT_LENGTH="90"

export TMUX_POWERLINE_WINDOW_STATUS_SEPARATOR=""

# Override theme left/right segments and colors
TMUX_POWERLINE_LEFT_STATUS_SEGMENTS=(
	"tmux_session_info 148 234"
)
TMUX_POWERLINE_RIGHT_STATUS_SEGMENTS=(
	"tmux_mem_cpu_load 234 136"
	"date_day 235 136"
	"date 235 136 ${TMUX_POWERLINE_SEPARATOR_LEFT_THIN}"
	"time 235 136 ${TMUX_POWERLINE_SEPARATOR_LEFT_THIN}"
	"xkb_layout 125 117"
)
```

**Comment by xx4h** [comment, marked as answer]
Hi @mikehaertl

I can not reproduce this behaviour with the changes you mentioned in `config.sh`.

To further investigate your issue, i can manually reproduce your behaviour by setting one of the following (not 100% sure your screenshot shows "black" or "transparent" background color):
- manually set `status-style` to "black" or "transparent":
  ```
  tmux set-option -g status-style bg=colour0
  ```
  (or `tmux set-option -g status-style default` for transparent)
- Set `status-style` in `~/.tmux.conf` after loading tmux-powerline:
  ```
  set -g status-style bg=colour0
  ```
- Set `TMUX_POWERLINE_STATUS_STYLE` in `config.sh`:
  ```
  export TMUX_POWERLINE_STATUS_STYLE="bg=colour0"
  ```

To see if `status-style` is the problem you can run
```
tmux show-options -g status-style
```
in your running tmux session and check if output looks like:
```
status-style fg=colour255,bg=colour235
```

Besides that i don't have any concrete idea right now, but if you can provide some more information, i'll have another look.

### Details

> I couldn't find a relevant configuration. In the `default.sh` theme file a `TMUX_POWERLINE_DEFAULT_BACKGROUND_COLOR` is set. But that's only used for the middle part (the window list).

The `TMUX_POWERLINE_DEFAULT_BACKGROUND_COLOR` is being used in multiple other locations
- in `lib/config_file.sh` in function `process_settings` function to set default value for `TMUX_POWERLINE_STATUS_STYLE`, which is being used tmux internally for `#[default]`
https://github.com/erikw/tmux-powerline/blob/717688b1dd9ca3bf062b44a59fb2206e07e28418/lib/config_file.sh#L73-L82
-  in `lib/powerline.sh` in function `__process_colors` function to set multiple default values for segment coloring
https://github.com/erikw/tmux-powerline/blob/717688b1dd9ca3bf062b44a59fb2206e07e28418/lib/powerline.sh#L134-L178
- in `lib/powerline.sh` in function `format` which is being used in the `default.sh` theme, as well as for the hardcoded defaults for `TMUX_POWERLINE_WINDOW_STATUS_CURRENT` and `TMUX_POWERLINE_WINDOW_STATUS_FORMAT`.
https://github.com/erikw/tmux-powerline/blob/717688b1dd9ca3bf062b44a59fb2206e07e28418/lib/powerline.sh#L51-L68

**Comment by mikehaertl** [comment]
@xx4h Thanks to your suggestions I found it. I had these hidden somewhere in my age old `.tmux.conf`:
```
# Set status line bg/fg colors
set-option -g status-bg black
set-option -g status-fg white
```

After removing them and restarting tmux everything looks good now 👍🏻 .

Thanks a lot!
